src/test/java/com/es/phoneshop/web/ProductDetailsPageServletTest.java

```diff
@@ -0,0 +1,75 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.dao.ProductDao;
+import com.es.phoneshop.exceptions.ProductNotFoundException;
+import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.service.ProductService;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.math.BigDecimal;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+
+public class ProductDetailsPageServletTest {
+
+    @Mock
+    private ProductService productService;
+
+    @Mock
+    private ProductDao productDao;
```

Comment:
why do we need dao in servlet? ProductDetailsPageServlet has only service as a resource